### DISCUSSION TITLE
3.6.1

### DISCUSSION THREAD

**Discussion by slorber** [facebook/docusaurus#10660] [Release feedback]
#### :bug: Bug Fix

- `docusaurus`
  - [#10658](https://github.com/facebook/docusaurus/pull/10658) fix(core): bundler should not minimize static assets ([@slorber](https://github.com/slorber))
- `docusaurus-bundler`, `docusaurus-faster`, `docusaurus-utils-common`, `docusaurus-utils`
  - [#10649](https://github.com/facebook/docusaurus/pull/10649) fix(faster,utils): fix faster/types peerDependencies ([@slorber](https://github.com/slorber))

#### :nail_care: Polish

- `docusaurus-bundler`, `docusaurus-types`, `docusaurus`
  - [#10655](https://github.com/facebook/docusaurus/pull/10655) refactor(faster,bundler,core): improve js loader DX ([@slorber](https://github.com/slorber))

#### :memo: Documentation

- [#10657](https://github.com/facebook/docusaurus/pull/10657) docs: fix old base ts config ref ([@slorber](https://github.com/slorber))

#### :wrench: Maintenance

- `docusaurus-mdx-loader`
  - [#10651](https://github.com/facebook/docusaurus/pull/10651) refactor(mdx-loader): streamline typescript usage for remark plugin types ([@lebalz](https://github.com/lebalz))
- Other
  - [#10650](https://github.com/facebook/docusaurus/pull/10650) chore: Argos screenshot dogfooding test pages ([@slorber](https://github.com/slorber))

#### Committers: 2

- Balthasar Hofer ([@lebalz](https://github.com/lebalz))
- Sébastien Lorber ([@slorber](https://github.com/slorber))

<hr /><em>This discussion was created from the release <a href='https://github.com/facebook/docusaurus/releases/tag/v3.6.1'>3.6.1</a>.</em>

**Comment by pranabdas** [comment]
@slorber the closing `</body>` and `</html>` tags are omitted from the generated HTML (it's still valid HTML as far as the browser is concerned). Wondering if it's intentional or overlooked 🤔

```console
curl -s https://docusaurus.io/index.html | tail -1
```
```
[snipped]
</a></div><div class=footer__copyright>Copyright © 2024 Meta Platforms,
 Inc. Built with Docusaurus.</div></div></div></footer></div>
```

**Reply by slorber** [reply]
Hmmm thanks, didn't notice that!

It seems like this is safe and done by our HTML minifier.

Previously we used https://github.com/terser/html-minifier-terser but had the `removeOptionalTags` option turned off.

Docusaurus Faster use the SWC HTML minifier: https://github.com/swc-project/swc/discussions/9616 and apparently does this by default. I'm not sure which option could disable it.

Has this been a problem in practice? I could look into it but wonder if it's worth it if things keep working.

**Reply by pranabdas** [reply]
It is practically safe, and the browser reconstructs the correct HTML. You can leave it as is, especially since it is the SWC default.

**Reply by slorber** [reply]
Yes we'll keep it this way for now, and see if anyone come up with a good reason to remove it.
They can temporarily switch back to the older HTML minifier if needed, so Docusaurus Faster is also an opportunity for us to be more aggressive in terms of potentially unsafe optimizations since they are all opt-in.

**Comment by hrumhurum** [comment]
I tried 3.6.1 with Docusaurus Faster enabled on several projects. The results are impressive:

- Everything works
- The build time is 5x-15x times faster depending on a project
- Wow

Thank you.

**Reply by slorber** [reply]
Awesome, thanks for reporting 🥳
